test/runtests.jl

```diff
@@ -143,6 +149,72 @@ end
     @test nvariables(H) == 3
 end
 
+
+@testset "GeodesicOnTheSphere" begin
+    @polyvar x y z
+    f = x + y + 3z
+    g = x + 2z
+    h = y + 2z
+
+
+    @test GeodesicOnTheSphere(f, g) isa GeodesicOnTheSphere{Float64}
+    @test GeodesicOnTheSphere([f, f], [g, h]) isa GeodesicOnTheSphere{Float64}
+    @test GeodesicOnTheSphere{Float64}([f, g], [f, h]) isa GeodesicOnTheSphere{Float64}
+    @test GeodesicOnTheSphere{Float64}(f, g) isa GeodesicOnTheSphere{Float64}
+    @test GeodesicOnTheSphere([FP.Polynomial(f)], [FP.Polynomial{Complex128}(f)]) isa GeodesicOnTheSphere{Complex128}
+    @test_throws AssertionError GeodesicOnTheSphere{Complex128}([f], [g])
+    @test_throws AssertionError GeodesicOnTheSphere{Complex128}([f], [f, f])
+
+
+    H = GeodesicOnTheSphere{Float64}([f, g], [f, h])
+    K = GeodesicOnTheSphere([f, g], [f, h])
+    @test promote_type(typeof(H), typeof(K)) == GeodesicOnTheSphere{Float64}
+    @test convert(GeodesicOnTheSphere{Float64}, K) isa GeodesicOnTheSphere{Float64}
+
+
+    w = [1.0, 2.0, 2.0]
+    @test evaluate(H, w, 1.0) == [9/4, 5/4]
+    @test H(w, 1.0) == [9/4, 5/4]
+    u = zeros(2)
+    evaluate!(u, H, w, 1.0)
+    @test u == [9/4, 5/4]
+
+
+    JH = jacobian(H)
+    @test JH(rand(3), 0.0) == [1 1 3; 0 1 2]./4
+    @test JH(rand(3), 1.0) == [1 1 3; 1 0 2]./4
+    u = zeros(2, 3)
+    JH! = jacobian!(H)
+    @test JH!(u, rand(3), 0.0) == JH(rand(3), 0.0)
+    @test u == JH(rand(3), 0.0)
+
+
+    HDT = dt(H)
+    @test HDT([1, 2, 2.0], 0.0) == [0, 0]
+    u = zeros(2)
+    HDT! = dt!(H)
+    HDT!(u, [1, 2, 2.0], 0.0)
+    @test u == [0, 0]
+    @test string(H) == "Homotopy.GeodesicOnTheSphere{Float64} The homotopy is given by the spherical geodesic from start/|start| (t=1) to target/|target| (t=0). \n"
+
+    N=weylnorm(H)
+    @test N(0.0)==1.0
+    @test N(0.5)==1.0
```

Comment:
Can you add a test for complex `t` and check the return type, with `@test N(t) isa Complex128`